File: 1380-lucky-numbers-in-a-matrix/1380.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        row = []
        col = []
        
        for r in matrix:
            row.append(min(r))
        
        for j in range(len(matrix[0])):
            maximum = matrix[0][j]
            for i in range(len(matrix)):
                maximum = max(matrix[i][j], maximum)
            col.append(maximum)
        
        ans = []
        for r in row:
            if r in col:
                ans.append(r)
        
        return ans

File: 1380-lucky-numbers-in-a-matrix/1380.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        set<int> row;
        vector<int> ans;
        for (int i = 0; i < matrix.size(); ++i) {
            row.insert(*min_element(matrix[i].begin(), matrix[i].end()));
        }
        for (int i = 0; i < matrix[0].size(); ++i) {
            int maximum = INT_MIN;
            for (int j = 0; j < matrix.size(); ++j) {
                maximum = max(maximum, matrix[j][i]);
            }
            if (row.find(maximum) != row.end())
                ans.push_back(maximum);
        }

        return ans;
    }
};
